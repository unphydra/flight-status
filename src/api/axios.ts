import axios from "axios";

export default axios.create({
  baseURL: "https://flight-status-mock.core.travelopia.cloud",
});
