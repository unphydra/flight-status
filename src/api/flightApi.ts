import { AxiosInstance } from "axios";

export enum FlightStatus {
  ON_TIME = "On Time",
  BOARDING = "Boarding",
  DELAYED = "Delayed",
  DEPARTED = "Departed",
}

export type FlightDetail = {
  id: number;
  flightNumber: string;
  airline: string;
  origin: string;
  destination: string;
  departureTime: string;
  status: FlightStatus;
};

export default class FlightApi {
  constructor(private apiService: AxiosInstance) {}

  async fetchFlights(): Promise<FlightDetail[]> {
    const { data } = await this.apiService.get("/flights");
    return data;
  }
}
