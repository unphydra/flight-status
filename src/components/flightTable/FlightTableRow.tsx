import { TableCell } from "@mui/material";
import TableRow from "@mui/material/TableRow";
import { FlightDetail } from "../../api/flightApi";

export function FlightTableRow({
  flightDetails,
}: {
  flightDetails: FlightDetail;
}) {
  return (
    <TableRow key={flightDetails.id}>
      <TableCell>{flightDetails.flightNumber}</TableCell>
      <TableCell>{flightDetails.airline}</TableCell>
      <TableCell>{flightDetails.origin}</TableCell>
      <TableCell>{flightDetails.destination}</TableCell>
      <TableCell>{flightDetails.departureTime}</TableCell>
      <TableCell>{flightDetails.status}</TableCell>
    </TableRow>
  );
}
