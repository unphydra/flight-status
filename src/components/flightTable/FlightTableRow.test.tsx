import { render } from "@testing-library/react";
import { FlightDetail, FlightStatus } from "../../api/flightApi";
import { FlightTableRow } from "./FlightTableRow";

describe("FlightTableRow", () => {
  const flightDetails: FlightDetail = {
    id: 1,
    flightNumber: "1",
    airline: "airline",
    origin: "origin",
    destination: "destination",
    departureTime: "departureTime",
    status: FlightStatus.ON_TIME,
  };
  it("Should renders without crashing", () => {
    const { container } = render(
      <FlightTableRow flightDetails={flightDetails} />
    );
    expect(container).toMatchSnapshot();
  });
});
