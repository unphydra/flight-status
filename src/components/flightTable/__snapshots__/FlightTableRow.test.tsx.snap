// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FlightTableRow > Should renders without crashing 1`] = `
<div>
  <tr
    class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
  >
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      1
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      airline
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      origin
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      destination
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      departureTime
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      On Time
    </td>
  </tr>
</div>
`;
